### DISCUSSION TITLE
Does Google provider support multi-tenancy ?

### DISCUSSION THREAD

**Discussion by KVRA** [nextauthjs/next-auth#4799] [Help]
### Question 💬

What I'm trying to do : I want to use the Google Provider but setting the "tenantID" ( don't exist ) 

Why: I need a list of users and db for production and another list of allowed users for staging, my manager told me to use "multi-tenancy" ( never used before )

What I tried to do : I see that other providers do have a parameter for "tenantId" but I cannot find it in Google Provider, also the google cloud documentation gives the example using the firebase sdk which wasn't helpfull since I'm using NextAuth
https://cloud.google.com/identity-platform/docs/multi-tenancy-authentication

Can I use NextAuth for this or I need to use firebase ? 

( Sorry to bother )

### How to reproduce ☕️

Google Provider don't have the parameter "tenantId" but other providers do have it and Google CLoud also supports multi-tenancy

### Contributing 🙌🏽

Yes, I am willing to help answer this question in a PR

**Comment by desisiv** [comment]
I have this same problem and haven't had any luck finding a solution.  Wondering if Google only supports it on the firebase sdk as you mention above.

**Reply by rogueturnip** [reply]
I was looking for this myself but appears the Google Provider in Next Auth is for signing in with a google account and not for connecting to Google Identity Platform which provides multi tenant support.  Reading the docs I don't see Google Identity Provider actually exposing OAuth or OIDC interfaces to support this integration either.

Anyone else seen similar or have a work around?  It would be nice to set this up with Next Auth rather than building custom (which I'm doing now).

**Reply by rockgabi** [reply]
Did you find any solution using next auth and google's identity platform + multi-tenancy?

**Reply by rogueturnip** [reply]
I didn't.  I ended writing the code myself including custom email delivery for forgot password and verify email since you can't customize in the console directly for google identity platform.

Doing this was a fair bit of code, having next-auth doing this would be nice but it's not impossible to do it yourself.  

One thing of note, I'm starting to look at Kinde Auth (https://kinde.com).  It is sort of like Clerk but has multi-tenancy support, larger set of stacks it supports and covers most of what Google had.  It's way easier to setup so I may swap there to reduce code I have to maintain.

**Comment by aliesenli** [comment]
Any news here, if `Google's Identity Platform` will be supported within next-auth anytime?

**Comment by ebratz** [comment]
I think the Google Provider will only handle the authentication part for you. I just had this on my side project and the way I handled it was writing an Adapter extending the standard Prisma Adapter (Which I was using) to include more columns to host Organizations or TenantID's in this case. You can see some examples on [this](https://github.com/nextauthjs/next-auth/discussions/11521) discussion I've opened and closed. Hope it brings some light to your needs
